core/src/main/scala/org/apache/spark/scheduler/AsyncEventQueue.scala

```diff
@@ -167,20 +170,27 @@ private class AsyncEventQueue(
     }
     logTrace(s"Dropping event $event")
 
-    val droppedCount = droppedEventsCounter.get
+    val droppedCount = droppedEventsCounter.get - lastDroppedEventsCounter
+    val lastReportTime = lastReportTimestamp.get
+    val curTime = System.currentTimeMillis()
     if (droppedCount > 0) {
       // Don't log too frequently
-      if (System.currentTimeMillis() - lastReportTimestamp >= 60 * 1000) {
-        // There may be multiple threads trying to decrease droppedEventsCounter.
-        // Use "compareAndSet" to make sure only one thread can win.
-        // And if another thread is increasing droppedEventsCounter, "compareAndSet" will fail and
-        // then that thread will update it.
-        if (droppedEventsCounter.compareAndSet(droppedCount, 0)) {
-          val prevLastReportTimestamp = lastReportTimestamp
-          lastReportTimestamp = System.currentTimeMillis()
-          val previous = new java.util.Date(prevLastReportTimestamp)
+      if (curTime - lastReportTime >= LOGGING_INTERVAL) {
+        // There may be multiple threads trying to logging dropped events,
+        // Use 'compareAndSet' to make sure only one thread can win.
+        if (lastReportTimestamp.compareAndSet(lastReportTime, curTime)) {
+          val lastReportTime = lastReportTimestamp.get
```

Comment:
Actually this is masking `lastReportTime` from outside the condition, and making the log always print the current time instead of the intended timestamp...

Reply:
Nice finding. Looks like this line should be removed.

Reply:
Good catching, I'll update it.